var/spack/repos/builtin/packages/neovim/package.py

```diff
@@ -7,34 +7,47 @@
 from spack import *
 
 
-class Neovim(CMakePackage):
-    """NeoVim: the future of vim"""
+class Neovim(MakefilePackage):
+    """Neovim: Vim-fork focused on extensibility and usability"""
```

Comment:
Why turn it into a `Makefile` package? The README says:
```
The build is CMake-based, but a Makefile is provided as a convenience
```
Since we have already a CMake based package I would keep it like that, but I'm curious if there are reasons I'm missing for this change.

Reply:
For convenience and simplification.

neovim [forks dependencies: libuv, libvterm, and unibilium](https://github.com/neovim/neovim/wiki/Deps/5e942d01eba237289e08d7472526169d58a9c2b1); especially, unibilium was abandoned.
These libraries are automatically built by default `make`.

Reply:
@adamjstewart What would be the best policy on the above? Usually we don't want packages to vendor their own dependencies, but:
1. `neovim` seems a final application (i.e. no other package is likely to use it as a dependency)
2. One of the libraries is a vendored fork with modifications

@e-kwsm What would we need to use CMake directly instead of a Makefile that uses CMake that generates another Makefile?

Reply:
`make` performs something like:
```sh
# @run_before('cmake')
cmake -B .deps -G Ninja third-party  # automatically detect Ninja
ninja -C .deps
# cmake
cmake -B build -G Ninja -DCMAKE_BUILD_TYPE=Debug .
# build
ninja -C build
```

`@run_before('cmake')` hook is used to mimic.